### PR TITLE
Aggregate linters for golangci and setup git hooks

### PR DESCRIPTION
**Pull request by rexagod**
Added linters to the golangci configuration and made respective changes
according to the resulting lint errors.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

### CODE REVIEW

**Comment by rexagod** [comment]
cc @aspandey @v-harihar

File: controllers/managedmcg_controller.go [hunk 1]
```diff
@@ -143,12 +140,14 @@ func (r *ManagedMCGReconciler) Reconcile(_ context.Context, req ctrl.Request) (c
...
 		statusErr = r.Client.Status().Update(r.ctx, r.managedMCG)
 	}
+
 	// Reconcile errors have priority to status update errors
-	if err != nil {
-		return ctrl.Result{}, err
-	} else if statusErr != nil {
-		return ctrl.Result{}, statusErr
-	} else {
+	switch {
```
Comment (naveenpaul1) [inline controllers/managedmcg_controller.go:145]: By replacing elseif with switch what we are trying to achieve here?

File: controllers/managedmcg_controller.go [hunk 2]
```diff
@@ -488,67 +504,79 @@ func (r *ManagedMCGReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
 	if err := r.list(&csvList); err != nil {
-		return nil, fmt.Errorf("unable to list csv resources: %v", err)
+		return nil, fmt.Errorf("unable to list csv resources: %w", err)
 	}
-	var csv *opv1a1.ClusterServiceVersion = nil
+	var csv *opv1a1.ClusterServiceVersion
 	for _, candidate := range csvList.Items {
 		if strings.HasPrefix(candidate.Name, name) {
-			csv = &candidate
+			candidateCSV := candidate
```
Comment (naveenpaul1) [inline controllers/managedmcg_controller.go:571]: Dose this make sense?

File: controllers/managedmcg_controller.go (hunk 1 above)
Comment (rexagod) [reply controllers/managedmcg_controller.go:145]: This change is suggested by the `gocognit` linter. PTAL https://github.com/uudashr/gocognit#understanding-the-complexity.

File: controllers/managedmcg_controller.go (hunk 2 above)
Comment (rexagod) [reply controllers/managedmcg_controller.go]: Fixed. Actually one of the linters had a problem with this due to the fact that the address will always be the same (only the value for the second for variable will change).

I've fixed this by using the first for key in the method, PTAL: https://stackoverflow.com/a/62486667.

A bit more on the explaination (copied from the first comment on the SO thread above):
```
Your for loop uses just 1 variable v and the value of v is updated by the for loop. But as it is just one variable it has just one address: &v is constant and does not change during the loop iteration. This is a typical problem in concurrent code as all created worker will work on the same memory location and thus either on the same value or on a value which is racy.
```

**Review by naveenpaul1 — CHANGES_REQUESTED** [review]
LGTM

**Review by naveenpaul1 — APPROVED** [review]

**Pull request merged**